src/Core/Banshee.WebBrowser/Banshee.WebSource/WebView.cs

```diff
@@ -26,64 +26,89 @@
-
-        public event EventHandler Ready;
+        private double smooth_scroll_size;
 
         public WebView ()
         {
             CanSearch = false;
         }
 
-        const float ZOOM_STEP = 0.05f;
+        protected string FixupJavascriptUrl { get; set; }
+
+        public bool IsReady { get; private set; }
+        public bool CanSearch { get; protected set; }
+
+        public event EventHandler Ready;
 
         public void ZoomIn ()
         {
-            Zoom += ZOOM_STEP;
+            Zoom += ZoomStep;
         }
 
         public void ZoomOut ()
         {
-            Zoom -= ZOOM_STEP;
+            Zoom -= ZoomStep;
         }
 
-        protected override bool OnScrollEvent (Gdk.EventScroll scroll)
+        [DllImport ("libgdk-3-0.dll", CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr gdk_event_get_scroll_deltas (IntPtr eventHandle, out double deltaX,
+            out double deltaY);
+
+        protected override bool OnScrollEvent (EventScroll scroll)
         {
-            if ((scroll.State & Gdk.ModifierType.ControlMask) != 0) {
-                Zoom += (scroll.Direction == Gdk.ScrollDirection.Up) ? ZOOM_STEP : -ZOOM_STEP;
+            if ((scroll.State & ModifierType.ControlMask) != 0) {
+                switch (scroll.Direction) {
+                    case ScrollDirection.Up:
+                        ZoomIn ();
+                        break;
+                    case ScrollDirection.Down:
+                        ZoomOut ();
+                        break;
+                    case (ScrollDirection) 4: //GDK_SCROLL_SMOOTH
+                        double delta_x;
+                        double delta_y;
+                        gdk_event_get_scroll_deltas (scroll.Handle, out delta_x, out delta_y);
+                        smooth_scroll_size += delta_y;
+
+                        if (smooth_scroll_size < -1) {
+                            ZoomIn ();
+                            smooth_scroll_size += 1;
+                        }
+                        else if (smooth_scroll_size > 1) {
+                            ZoomOut ();
+                            smooth_scroll_size -= 1;
+                        }
+                        break;
+                    default:
+                        return true;
+                }
                 return true;
             }
-
             return base.OnScrollEvent (scroll);
         }
 
         protected override void OnLoadStatusChanged (OssiferLoadStatus status)
         {
             if ((status == OssiferLoadStatus.FirstVisuallyNonEmptyLayout ||
-                status == OssiferLoadStatus.Finished) && Uri != "about:blank") {
-                if (fixup_javascript != null) {
-                    ExecuteScript (fixup_javascript);
-                }
-            }
+                 status == OssiferLoadStatus.Finished) &&
+                Uri != "about:blank") if (fixup_javascript != null) ExecuteScript (fixup_javascript);
```

Comment:
definitely not a fan of this formatting, I'm a braces kinda guy 😆

I wasn't a fan of the prior formatting either though, is `status` an enum here? -  I'm wondering if we can check the `Uri` and `fixup_javascript` conditions first and then switch on `status` or if you don't like the switch then we've only got two conditions so it's probably okay to bend the style rules a little with a long line here. (just fixup the commit where you did the formatting, it's the same action)

Reply:
Oh my... That looks horrible. No idea how I missed that. It might have been late?!

The logic part might change in the near future (the webkit PR). The `fixup_javascript` might not be needed anymore.